daemon/images/image_pull.go

```diff
@@ -50,6 +53,28 @@ func (i *ImageService) PullImage(ctx context.Context, image, tag string, platfor
 
 	err = i.pullImageWithReference(ctx, ref, platform, metaHeaders, authConfig, outStream)
 	imageActions.WithValues("pull").UpdateSince(start)
+
+	if platform != nil {
+		img, err := i.GetImage(image, platform)
+		if errdefs.IsNotFound(err) && img != nil {
+			imgPlat := specs.Platform{
+				OS:           img.OS,
+				Architecture: img.BaseImgArch(),
+				Variant:      img.BaseImgVariant(),
+			}
+
+			p := *platform
+			if !platforms.Only(p).Match(imgPlat) {
+				po := streamformatter.NewJSONProgressOutput(outStream, false)
+				progress.Messagef(po, "", `WARNING: Pulled image with specified platform (%s), but the resulting image's configured platform (%s) does not match.`,
+					platforms.Format(p), platforms.Format(imgPlat),
+				)
+				logrus.WithError(err).WithField("image", image).Warn("Ignoring error about platform mismatch where the manifest list points to an image whose configuration does not match the platform in the manifest.")
```

Comment:
Let me fix this log, as it's mentioning `manifest list` which is not the case here

Reply:
Oh, actually, `err` already contains all info we need;

```
image with reference armhf/busybox was found but does not match the specified platform: wanted linux/s390x, actual: linux/arm64
```

We can probably just use the error